neural_lam/models/ar_model.py

```diff
@@ -58,21 +91,30 @@ def __init__(self, args):
             self.num_grid_nodes,
             grid_static_dim,
         ) = self.grid_static_features.shape
+
         self.grid_dim = (
-            2 * self.config_loader.num_data_vars()
+            2 * self.grid_output_dim
             + grid_static_dim
-            + self.config_loader.dataset.num_forcing_features
+            + num_forcing_vars * forcing_window_size
```

Comment:
This is the only place where `forcing_window_size` is used, but we could just as well use `args.forcing_window_size`. That way there is no need to pass `forcing_window_size` as a positional argument when instantiating every model class (it's inconsistent why we would include this explicitly in the constructor when we don't do it with anything else in `args`).

Reply:
I addressed this in https://github.com/leifdenby/neural-lam/commit/6f1efd657e76fa1290b33d671c2910cf42602e46, https://github.com/leifdenby/neural-lam/commit/dba94b33eba80f64088a8660c275898d1b80a529